csp_billing_adapter_google/plugin.py

```diff
@@ -37,10 +45,71 @@ def setup_adapter(config: Config):
 def meter_billing(
     config: Config,
     dimensions: dict,
-    timestamp: str,
+    timestamp: datetime,
     dry_run: bool
 ) -> dict:
-    return {}
+    """
+    Process a metered billing based on the dimensions provided
+
+    Uses google ubbagent sidecare
+    If a single dimension is provided the meter_usage API is
+    used for the metering. If there is an error the metering
+    is attempted 3 times before re-raising the exception to
+    calling scope.
+    """
+    retries = 3
+    status = {}
+
+    for dimension_name, usage_quantity in dimensions.items():
+        body = {
+            "name": dimension_name,
+            "startTime": str(timestamp),
+            "endTime":  str(timestamp),
+            "value": {
+                'int64value': usage_quantity,
+                },
+            }
+        uubody = json.dumps(body)
+        req = urllib.request.Request(
+            UBBA_REPORT_URL,
+            data=uubody.encode(),
+            headers={
+                'Content-type': 'application/json'
+                },
+            method='POST'
+            )
+
+        retries = 3
```

Comment:
There is a retry on exception helper script implemented in the core adapter if you want to leverage that instead?

https://github.com/SUSE-Enceladus/csp-billing-adapter/blob/f6a10a62fa99db0de58cf6bff58a2c09806eb89d/csp_billing_adapter/utils.py#L141